Integrations/src/main/java/io/deephaven/integrations/learn/Gatherer.java

```diff
@@ -0,0 +1,223 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.v2.sources.ColumnSource;
+
+public class Gatherer {
+
+    /**
+     * Copy data from a table into a 2d tensor of Booleans.
+     *
+     * @param indexSet indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
+     *         handle passing the memory reference as a 1d java array here.
+     */
+
+    public static Boolean[] booleanTensorBuffer2D(final IndexSet indexSet, final ColumnSource<?>[] columnSources) {
+
+        final int nRows = indexSet.getSize();
+        final int nCols = columnSources.length;
+        final Boolean[] tensor = new Boolean[nRows * nCols];
+
+        int j = 0;
+        for (ColumnSource<?> cs : columnSources) {
+            int i = 0;
+
+            for (long idx : indexSet) {
+                final int ij = nCols * i + j;
+                tensor[ij] = cs.getBoolean(idx);
+                i++;
+            }
+
+            j++;
+        }
+
+        return tensor;
+    }
+
+    /**
+     * Copy data from a table into a 2d tensor of bytes.
+     *
+     * @param indexSet indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
+     *         handle passing the memory reference as a 1d java array here.
+     */
+
+    public static byte[] byteTensorBuffer2D(final IndexSet indexSet, final ColumnSource<?>[] columnSources) {
+
+        final int nRows = indexSet.getSize();
+        final int nCols = columnSources.length;
+        final byte[] tensor = new byte[nRows * nCols];
+
+        int j = 0;
+        for (ColumnSource<?> cs : columnSources) {
+            int i = 0;
+
+            for (long idx : indexSet) {
+                final int ij = nCols * i + j;
+                tensor[ij] = cs.getByte(idx);
```

Comment:
For `NULL_<TYPE>` values, should we offer an option to (1) do what we do here, (2) raise an exception or (3) map to a value?

Reply:
I'm not sure how we want to do null handling.  I didn't include it on this update, though it can be added soon

Reply:
It seems like there are a few options:
1) just cram our null values into the tensor.  This will be fast but relies on the user to have thought out how they want nulls to behave.
2) raise an exception on null values to force the user to deal with them before this function.
3) provide a default value to use for null values.  Somewhat of a compromise.

The three options are not exclusive.  The function could potentially support them all reasonably efficiently if you are careful.  Thoughts on which ones we should support?

Reply:
To easily see what is currently going on with nulls, write a unit test.